src/proxy/http/metrics/report.rs

```diff
@@ -89,6 +90,9 @@ where
         self.scope.response_total().fmt_help(f)?;
         registry.fmt_by_class(f, self.scope.response_total(), |s| &s.total)?;
 
+        self.scope.retry_skipped_total().fmt_help(f)?;
+        registry.fmt_by_retry(f, self.scope.retry_skipped_total())?;
```

Comment:
As this is, we'll report these metrics for all http scopes, which is a bit confusing.  Perhaps the retry-specific stats logic should just be embedded directly in the retries module? something to the effect of having `retries::new() -> (Layer, Report)` ?

We would probably end up with keys like
- `route_effective_...` -- http metrics
- `route_retry_(count|skipped|budget)` -- retry metrics 
- `route_absolute_...` -- http metrics